### DISCUSSION TITLE
window.innerWidth / innerHeight

### DISCUSSION THREAD

**Discussion by WebMechanic** [capricorn86/happy-dom#591] [Q&A] [answered]
Hello,

up until v6 the size of the window object via innerWidth / innerHeight was reported as 1024 * 768. 
In v7 they are 0 incl. all offset* properties of the document and documentElement.

a) Was this change deliberate with the new addition of the various CSS interfaces?
b) How can I change the "size" of the window so there is something "to work with"?
c) Is the some documentation about the implementation levels of the various DOM APIs?

Thank you.

**Comment by WebMechanic** [comment]
I found this strolling the repo ...
https://github.com/capricorn86/happy-dom/tree/master/packages/happy-dom#additional-features

window.happyDOM.setInnerWidth(1024);
window.happyDOM.setInnerHeight(768);

**Comment by capricorn86** [comment, marked as answer]
Hi @WebMechanic! 🙂 

The change was intentional as it is now possible to send in "innerWidth" and "innerHeight" as properties to the constructor.

Offset properties has not been changed. They are set to 0 as Happy DOM currently doesn't support calculating them yet, but I guess we could set them to the same as "innerWidth" and "innerHeight" on the document element.

It is also possible to set "innerWidth" and "innerHeight" now through a method, which will trigger a "resize" event.

I have just updated the documentation to cover this.

Documentation:
https://github.com/capricorn86/happy-dom/tree/master/packages/happy-dom

Example 1:
```javascript
const window = new Window({
    innerWidth: 1024,
    innerHeight: 768,
    url: 'http://localhost:8080'
});
 ```
 
Example 2:
```javascript
window.happyDOM.setInnerWidth(1024);
window.happyDOM.setInnerHeight(1024);
 ```

**Reply by WebMechanic** [reply]
Hey, thanks. 
Guess we missed each other :)

Anything of point c) ?

**Reply by capricorn86** [reply]
Sorry, I missed that 😅 

Unfortunately there is no such document in place yet, but it would definitely be good to have. I would also like to improve the documentation in general to explain better how Happy DOM can be used.

**Reply by WebMechanic** [reply]
> Offset properties has not been changed. They are set to 0 as Happy DOM currently doesn't support calculating them yet, but I guess we could set them to the same as "innerWidth" and "innerHeight" on the document element.

maybe they could-should all default to/inherit the constants you already have in Screen.ts? 
That's what a browser would resolve to in fullscreen mode  -- would've done, back in 1996 on a modern CRT 😇

oh, and **thank you very much for making this**! 🚀

**Reply by csantos-nydig** [reply]
I would like to use `window.happyDOM.setInnerWidth(1024);` in my tests with **vitest**; but TypeScript is complaining:

<img width="716" alt="image" src="https://github.com/capricorn86/happy-dom/assets/87080313/4936ed60-adf5-431d-941f-ea283bc809d5">

any idea how to make TS happy?

**Reply by csantos-nydig** [reply]
Also, while at it: is there a way to "reset" it back to the default?

I'm currently doing this:

```tsx
it('test', () => {
  window.happyDOM.setViewport({ width: 100 });
  render(<TestMatchMedia />);
  window.happyDOM.setViewport({ width: 1024 });
});
```

but it feels wrong

**Reply by KevinGhadyani-Okta** [reply]
Does this only affect the window height? What's the trick to getting other elements sized up correctly?

If I do something as simple as set a CSS height to a fixed value, what tells Happy-DOM, how do I get Happy-DOM to report a height?
```ts
const window = new Window({
  innerWidth: 1024,
  innerHeight: 768,
});

const containerElement = window.document.createElement("div")

containerElement.style.setProperty("height", "100px")

window.document.body.append(containerElement)
```
